<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1001</width>
    <height>879</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>Web Browser</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../app/assets/application.qrc">
    <normaloff>:/logo.png</normaloff>:/logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="sizeConstraint">
     <enum>QLayout::SetDefaultConstraint</enum>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="BookmarkBar" name="bookmarkBar" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>30</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="FindTextWidget" name="widgetFindText" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1001</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew_Tab"/>
    <addaction name="actionNew_Window"/>
    <addaction name="actionNew_Private_Window"/>
    <addaction name="separator"/>
    <addaction name="actionOpen_File"/>
    <addaction name="actionClose_Tab"/>
    <addaction name="actionClose_Window"/>
    <addaction name="separator"/>
    <addaction name="action_Save_Page_As"/>
    <addaction name="separator"/>
    <addaction name="action_Print"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="action_Undo"/>
    <addaction name="action_Redo"/>
    <addaction name="separator"/>
    <addaction name="actionCu_t"/>
    <addaction name="action_Copy"/>
    <addaction name="action_Paste"/>
    <addaction name="action_Delete"/>
    <addaction name="separator"/>
    <addaction name="action_Find"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionBookmark_Bar"/>
    <addaction name="separator"/>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_Out"/>
    <addaction name="actionReset_Zoom"/>
    <addaction name="separator"/>
    <addaction name="actionReload"/>
    <addaction name="separator"/>
    <addaction name="actionPage_So_urce"/>
    <addaction name="action_Full_Screen"/>
   </widget>
   <widget class="HistoryMenu" name="menuHistory">
    <property name="title">
     <string>Histo&amp;ry</string>
    </property>
   </widget>
   <widget class="BookmarkMenu" name="menuBookmarks">
    <property name="styleSheet">
     <string notr="true">* { menu-scrollable: 1 }</string>
    </property>
    <property name="title">
     <string>Boo&amp;kmarks</string>
    </property>
   </widget>
   <widget class="ToolMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>He&amp;lp</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuHistory"/>
   <addaction name="menuBookmarks"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="NavigationToolBar" name="toolBar">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>36</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Show tool bar</string>
   </property>
   <property name="styleSheet">
    <string notr="true">QToolBar { spacing: 3px; } QToolButton::menu-indicator { subcontrol-position: bottom right; subcontrol-origin: content; }</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetClosable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew_Window">
   <property name="icon">
    <iconset theme="window-new">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;New Window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionNew_Tab">
   <property name="icon">
    <iconset theme="folder-new">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>New &amp;Tab</string>
   </property>
   <property name="iconText">
    <string>New Tab</string>
   </property>
   <property name="toolTip">
    <string>New Tab</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionNew_Private_Window">
   <property name="icon">
    <iconset theme="window-new">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>New Private &amp;Window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+P</string>
   </property>
  </action>
  <action name="actionOpen_File">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Open File...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClose_Tab">
   <property name="icon">
    <iconset theme="window-close">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Close Tab</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionClose_Window">
   <property name="icon">
    <iconset theme="window-close">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>C&amp;lose Window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+W</string>
   </property>
  </action>
  <action name="action_Save_Page_As">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Save Page As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_Print">
   <property name="icon">
    <iconset theme="document-print">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Print...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-undo">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-redo">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionCu_t">
   <property name="icon">
    <iconset theme="edit-cut">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="action_Copy">
   <property name="icon">
    <iconset theme="edit-copy">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_Paste">
   <property name="icon">
    <iconset theme="edit-paste">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="action_Delete">
   <property name="icon">
    <iconset theme="edit-delete">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_Find">
   <property name="icon">
    <iconset theme="edit-find">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Find</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="icon">
    <iconset theme="preferences-other">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Pr&amp;eferences</string>
   </property>
  </action>
  <action name="actionBookmark_Bar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="user-bookmarks">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Bookmark Bar</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="icon">
    <iconset theme="zoom-in">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="icon">
    <iconset theme="zoom-out">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Zoom &amp;Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionReset_Zoom">
   <property name="icon">
    <iconset theme="zoom-original">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Reset &amp;Zoom</string>
   </property>
  </action>
  <action name="actionPage_So_urce">
   <property name="icon">
    <iconset theme="accessories-text-editor">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Page So&amp;urce</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="action_Full_Screen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="view-fullscreen">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Full Screen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionManage_Cookies">
   <property name="text">
    <string>&amp;Manage Cookies</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About Viper Browser</string>
   </property>
  </action>
  <action name="actionShow_all_history">
   <property name="text">
    <string>&amp;Show All History</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+H</string>
   </property>
  </action>
  <action name="actionClear_Recent_History">
   <property name="text">
    <string>&amp;Clear Recent History...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Del</string>
   </property>
  </action>
  <action name="actionReload">
   <property name="icon">
    <iconset theme="view-refresh">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Reload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionView_Downloads">
   <property name="text">
    <string>&amp;View Downloads</string>
   </property>
  </action>
  <action name="actionDefault">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Default</string>
   </property>
  </action>
  <action name="actionAdd_User_Agent">
   <property name="text">
    <string>&amp;Add User Agent</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
  <action name="actionUser_Scripts">
   <property name="text">
    <string>Manage User &amp;Scripts</string>
   </property>
  </action>
  <action name="actionManage_Ad_Blocker">
   <property name="text">
    <string>Manage Ad &amp;Blocker</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FindTextWidget</class>
   <extends>QWidget</extends>
   <header>FindTextWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>BookmarkBar</class>
   <extends>QWidget</extends>
   <header>BookmarkBar.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HistoryMenu</class>
   <extends>QMenu</extends>
   <header>HistoryMenu.h</header>
  </customwidget>
  <customwidget>
   <class>BookmarkMenu</class>
   <extends>QMenu</extends>
   <header>BookmarkMenu.h</header>
  </customwidget>
  <customwidget>
   <class>NavigationToolBar</class>
   <extends>QToolBar</extends>
   <header>NavigationToolBar.h</header>
  </customwidget>
  <customwidget>
   <class>ToolMenu</class>
   <extends>QMenu</extends>
   <header>ToolMenu.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../app/assets/application.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionOpen_File</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openFileInBrowser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Print</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>printTabContents()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose_Window</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Find</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onFindTextAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Full_Screen</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onToggleFullScreen(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openPreferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPage_So_urce</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onRequestViewSource()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFileInBrowser()</slot>
  <slot>printTabContents()</slot>
  <slot>onFindTextAction()</slot>
  <slot>onToggleFullScreen(bool)</slot>
  <slot>openPreferences()</slot>
  <slot>onRequestViewSource()</slot>
 </slots>
</ui>
